tools/engine_tool/lib/src/commands/query_command.dart

```diff
@@ -0,0 +1,126 @@
+// found in the LICENSE file.
+
+import 'package:engine_build_configs/engine_build_configs.dart';
+
+import 'command.dart';
+
+const String _allFlag = 'all';
+const String _builderFlag = 'builder';
+const String _verboseFlag = 'verbose';
+
+/// The root 'query' command.
+final class QueryCommand extends CommandBase {
+  /// Constructs the 'query' command.
+  QueryCommand({
+    required super.environment,
+    required this.configs,
+  }) {
+    // Add options here that are common to all queries.
+    argParser
+      ..addFlag(
+        _allFlag,
+        abbr: 'a',
+        help: 'List all results, even when not relevant on this platform',
+        negatable: false,
+      )
+      ..addOption(
+        _builderFlag,
+        abbr: 'b',
+        help: 'Restrict the query to a single builder.',
+        allowed: <String>[
+          for (final MapEntry<String, BuildConfig> entry in configs.entries)
+            if (entry.value.canRunOn(environment.platform))
+              entry.key,
+        ],
+        allowedHelp: <String, String>{
+          // TODO(zanderso): Add human readable descriptions to the json files.
+          for (final MapEntry<String, BuildConfig> entry in configs.entries)
+            if (entry.value.canRunOn(environment.platform))
+              entry.key: entry.value.path,
+        },
+      )
+      ..addFlag(
+        _verboseFlag,
+        abbr: 'v',
+        help: 'Respond to queries with extra information',
+        negatable: false,
+      );
+
+    addSubcommand(QueryBuildsCommand(
+      environment: environment,
+      configs: configs,
+    ));
+  }
+
+  /// Build configurations loaded from the engine from under ci/builders.
+  final Map<String, BuildConfig> configs;
+
+  @override
+  String get name => 'query';
+
+  @override
+  String get description => 'Provides information about build configurations '
+                            'and tests.';
+
+  @override
+  Future<int> run() async {
+    environment.stdout.write(usage);
+    return 0;
+  }
+}
+
+/// The 'query builds' command.
+final class QueryBuildsCommand extends CommandBase {
+  /// Constructs the 'query build' command.
+  QueryBuildsCommand({
+    required super.environment,
+    required this.configs,
+  });
+
+  /// Build configurations loaded from the engine from under ci/builders.
+  final Map<String, BuildConfig> configs;
+
+  @override
+  String get name => 'builds';
+
+  @override
+  String get description => 'Provides information about CI build '
+                            'configurations';
+
+  @override
+  Future<int> run() async {
+    // Loop through all configs, and log those that are compatible with the
+    // current platform.
+    final bool all = parent!.argResults![_allFlag]! as bool;
+    final String? builderName = parent!.argResults![_builderFlag] as String?;
+    final bool verbose = parent!.argResults![_verboseFlag] as bool;
+    for (final String key in configs.keys) {
```

Comment:
In my opinion, this output is a bit too succinct by default. What do you think of outputting something like this?

```
Add --verbose to see builders' information

linux_clang_tidy config:
  android_debug_arm64 builder
  host_debug builder
linux_host_desktop_engine config:
  host_debug builder
  host_profile builder
  host_release builder
...
```

Reply:
Update with this. I'm hoping that we'll make this look even better (or at least more consistent) once we bring in a real terminal logging package.